modin/core/dataframe/algebra/default2pandas/groupby.py

```diff
@@ -491,9 +491,7 @@ def handle_as_index(
         if not keep_index_levels:
             # We want to insert only these internal-by-cols that are not presented
             # in the result in order to not create naming conflicts
-            if selection is None and internal_by_cols.nlevels != result_cols.nlevels:
-                cols_to_insert = internal_by_cols.copy()
-            elif selection is None:
```

Comment:
There is no way of getting a different amount of levels for the 'by' columns and for the resulting ones unless it's a renaming aggregation (handled separately in the `selection is not None` branch).

This if-branch was originally misported from the [`qc.groupby_agg`](https://github.com/modin-project/modin/blob/045eb9353f234f67e5bf35272ea98aac9c05cdf2/modin/core/storage_formats/pandas/query_compiler.py#L2764-L2775) where there was no distinguishing between `selection` and `non-selection` aggregations.